### PR TITLE
Bring rustql up to date with the latest compiler

### PR DESCRIPTION
**Pull request by ktrianta**
The main purpose of this PR is to bring rustql up to date with the latest changes in the nightly compiler interfaces, specifically **rustc 1.38.0-nightly (890881f8f 2019-07-25)**.

Most of the updates are implemented in the rustql-extractor and have to do with the rustc_driver and how it is invoked. The CrateVisitor has been also updated to account for the latest changes in the HIR, MIR and general internal compiler interfaces.

We also have updated all rustql crates to use the latest available versions of their dependencies.

### CODE REVIEW

File: rustql-extractor/Cargo.toml [hunk 1]
```diff
@@ -8,13 +8,14 @@ license = "MIT/Apache-2.0"
 edition = "2018"
 
 [[bin]]
-name = "rustc"
+name = "rustql-extractor"
```
Comment (vakaras) [inline rustql-extractor/Cargo.toml]: Have you tried running with [sccache](https://github.com/mozilla/sccache)? At least `sccache` use to assume that the binary is called `rustc`.

File: rustql-extractor/src/main.rs [hunk 2]
```diff
@@ -29,163 +31,198 @@ extern crate syntax;
...
-use self::rustc_driver::driver::{CompileState, CompileController};
-
 use rustql_common::data;
 
 //use rustc_driver::driver::{CompileController, PhaseController, CompileState};
 use crate::rustc::hir;
 use crate::rustc::hir::intravisit::walk_crate;
+use rustc_driver::Compilation;
+use rustc_interface::interface;
+use rustc_tools_util::*;
```
Comment (vakaras) [inline rustql-extractor/src/main.rs]: Is it necessary to have a wildcard import here?

File: rustql-extractor/src/main.rs [hunk 3]
```diff
@@ -29,163 +31,198 @@ extern crate syntax;
...
+                //println!("{:?}", cv.crate_data);
+                let result = export_crate(&cv.crate_data);
+                if let None = result {
+                    println!(
+                        "ERROR exporting crate: {}",
+                        cv.crate_data.metadata.get_filename()
+                    );
+                }
+            });
+        Compilation::Stop
```
Comment (vakaras) [inline rustql-extractor/src/main.rs]: What happens when the extractor is also invoked on dependencies? At least my understanding was that if we prematurely stop the compilation while compiling dependencies, then dependent crates cannot be compiled anymore.

**Review by vakaras — CHANGES_REQUESTED** [review]
Thank you for the pull request! I have added some minor comments. Once they are addressed, I will merge the pull request.

File: rustql-extractor/src/main.rs (hunk 2 above)
Comment (ktrianta) [reply rustql-extractor/src/main.rs]: Fixed.

File: rustql-extractor/Cargo.toml (hunk 1 above)
Comment (ktrianta) [reply rustql-extractor/Cargo.toml]: I hadn't tried that. I tried and is still needed so I changed it.

File: rustql-extractor/src/main.rs (hunk 3 above)
Comment (ktrianta) [reply rustql-extractor/src/main.rs]: You are right. I was not aware of that, but it is fixed now.

**Pull request merged**